packages/suite/src/views/wallet/tokens/common/TokensLayout/TokensLayoutNavigation.tsx

```diff
@@ -0,0 +1,119 @@
+import styled, { css } from 'styled-components';
+
+import { TranslationKey } from '@suite-common/intl-types';
+import { Route } from '@suite-common/suite-types';
+import { IconName } from '@suite-common/icons';
+import { SelectedAccountLoaded } from '@suite-common/wallet-types';
+import { selectCoinDefinitions } from '@suite-common/token-definitions';
+import { Elevation, borders, spacingsPx, mapElevationToBorder, typography } from '@trezor/theme';
+import { useElevation, variables } from '@trezor/components';
+
+import { useSelector } from 'src/hooks/suite';
+import { NavigationItem } from '../../../../../components/suite/layouts/SuiteLayout/Sidebar/NavigationItem';
+import { getTokens } from 'src/utils/wallet/tokenUtils';
+
+const List = styled.div`
+    display: flex;
+    gap: ${spacingsPx.xxs};
+    margin-bottom: ${spacingsPx.md};
+`;
+
+const NavListItem = styled(NavigationItem)`
+    gap: ${spacingsPx.xs};
+    padding: ${spacingsPx.xs} ${spacingsPx.md};
+    ${typography.hint}
+
+    ${({ isActive }) =>
+        isActive &&
+        css`
+            border-radius: ${borders.radii.full};
+        `}
+`;
+
+const StyledDivider = styled.div<{ $elevation: Elevation }>`
+    margin: ${spacingsPx.xs} ${spacingsPx.xxs};
+    border-left: 1px solid ${mapElevationToBorder};
+
+    ${variables.SCREEN_QUERY.BELOW_LAPTOP} {
+        margin: 0 ${spacingsPx.xxs};
+        flex-direction: row;
+        border-bottom: 0;
+        border-right: 1px solid ${mapElevationToBorder};
+    }
+`;
+
+const Divider = () => {
+    const { elevation } = useElevation();
+
+    return <StyledDivider $elevation={elevation} />;
+};
+
+interface TokensLayoutNavigationProps {
+    selectedAccount: SelectedAccountLoaded;
+}
+
+export const TokensLayoutNavigation = ({ selectedAccount }: TokensLayoutNavigationProps) => {
+    const routeName = useSelector(state => state.router.route?.name);
+    const coinDefinitions = useSelector(state =>
+        selectCoinDefinitions(state, selectedAccount.account.symbol),
+    );
+
+    const shownTokens = getTokens(
```

Comment:
Not strong opinion:
What if instead of calling `getTokens` 3 times
<img width="320" alt="Screenshot 2024-06-12 at 12 30 08" src="https://github.com/trezor/trezor-suite/assets/33235762/511652e4-7cf8-436f-804f-ac5953e8a101">
it would return object of 
{shows: [], hidden: [], unverified: []}